### ISSUE TITLE
How to use 3rd-party CSS?

### ISSUE THREAD

**Issue by neil-s** [zurb/foundation-cli#21]
I'm trying to use the Card component from Semantic UI (https://github.com/Semantic-Org/UI-Card), which is available both as a bower and npm package. Installing the package downloads a CSS file (card.css), which I want to include in my dist stylesheet. I tried doing this by:

1. `bower install --save semantic-ui-card`
1. In the gulfile.js, update the value of PATHS.sass to:
  
   ```
    sass: [
       'bower_components/foundation-sites/scss',
       'bower_components/motion-ui/src/',
       'bower_components/semantic-ui-card/card.css'
     ],
   ```
3. Running `gulp build`, `gulp watch` and `foundation watch`

I expected that this would download the css file, and include its contents in the dist output due to the 'sass' gulp task.

However, while the card.css file has been correctly downloaded to the path mentioned above, its contents are not included in `dist\assets\css\app.css`. 

I also tried changing the file extension to `.scss` , and updating the path accordingly. I tried installing by npm first and that didn't work either. 

Could you tell me how I'm meant to import 3rd-party css from the bower_components folder into my production code?

**Comment by gakimball** [comment]
Sass can't `@import` a plain CSS file, so you have to set things up a little differently.

After the Sass has compiled, you can insert new files into the Gulp stream, and then merge those new files with your initial CSS file.

Using the `sass` task from the ZURB Template, here's how you might augment it to support external CSS:

```js
var EXTERNAL_CSS = [
  'bower_components/semantic-ui-card/card.css'
];

return gulp.src('src/assets/scss/app.scss')
  .pipe($.sourcemaps.init())
  .pipe($.sass({
    includePaths: PATHS.sass
  }).on('error', $.sass.logError))
  // Add new files to the stream
  .pipe(gulp.src(EXTERNAL_CSS))
  // Combine the new files with the previous CSS
  .pipe($.concat('app.css'))
  // The rest of the stream works the same
  .pipe($.autoprefixer({
    browsers: COMPATIBILITY
  }))
  .pipe(uncss)
  .pipe(minifycss)
  .pipe($.if(!isProduction, $.sourcemaps.write()))
  .pipe(gulp.dest('dist/assets/css'));
```

Let me know if this helps!

**Comment by neil-s** [comment]
Thanks for the quick help @gakimball . I tried adding the lines you mentioned, but now the output app.css seems to *only* contain the contents of `card.css`, not Foundation or my custom `app.scss` code.

Copying my `gulpfile.js` in case I've made a transcription error I'm not spotting. And yes, you assumed correctly, I am using the Zurb template.

```
var $        = require('gulp-load-plugins')();
var argv     = require('yargs').argv;
var browser  = require('browser-sync');
var gulp     = require('gulp');
var panini   = require('panini');
var rimraf   = require('rimraf');
var sequence = require('run-sequence');
var sherpa   = require('style-sherpa');

// Check for --production flag
var isProduction = !!(argv.production);

// Port to use for the development server.
var PORT = 8000;

// Browsers to target when prefixing CSS.
var COMPATIBILITY = ['last 2 versions', 'ie >= 9'];

// File paths to various assets are defined here.
var PATHS = {
  external_css: [
    'bower_components/semantic-ui-card/card.css'
  ],
  assets: [
    'src/assets/**/*',
    '!src/assets/{!img,js,scss}/**/*'
  ],
  sass: [
    'bower_components/foundation-sites/scss',
    'bower_components/motion-ui/src/'
  ],
  javascript: [
    'bower_components/jquery/dist/jquery.js',
    'bower_components/what-input/what-input.js',
    'bower_components/foundation-sites/js/foundation.core.js',
    'bower_components/foundation-sites/js/foundation.util.*.js',
    // Paths to individual JS components defined below
    'bower_components/foundation-sites/js/foundation.abide.js',
    'bower_components/foundation-sites/js/foundation.accordion.js',
    'bower_components/foundation-sites/js/foundation.accordionMenu.js',
    'bower_components/foundation-sites/js/foundation.drilldown.js',
    'bower_components/foundation-sites/js/foundation.dropdown.js',
    'bower_components/foundation-sites/js/foundation.dropdownMenu.js',
    'bower_components/foundation-sites/js/foundation.equalizer.js',
    'bower_components/foundation-sites/js/foundation.interchange.js',
    'bower_components/foundation-sites/js/foundation.magellan.js',
    'bower_components/foundation-sites/js/foundation.offcanvas.js',
    'bower_components/foundation-sites/js/foundation.orbit.js',
    'bower_components/foundation-sites/js/foundation.responsiveMenu.js',
    'bower_components/foundation-sites/js/foundation.responsiveToggle.js',
    'bower_components/foundation-sites/js/foundation.reveal.js',
    'bower_components/foundation-sites/js/foundation.slider.js',
    'bower_components/foundation-sites/js/foundation.sticky.js',
    'bower_components/foundation-sites/js/foundation.tabs.js',
    'bower_components/foundation-sites/js/foundation.toggler.js',
    'bower_components/foundation-sites/js/foundation.tooltip.js',
    'src/assets/js/!(app.js)**/*.js',
    'src/assets/js/app.js'
  ]
};

// Delete the "dist" folder
// This happens every time a build starts
gulp.task('clean', function(done) {
  rimraf('dist', done);
});

// Copy files out of the assets folder
// This task skips over the "img", "js", and "scss" folders, which are parsed separately
gulp.task('copy', function() {
  gulp.src(PATHS.assets)
    .pipe(gulp.dest('dist/assets'));
});

// Copy page templates into finished HTML files
gulp.task('pages', function() {
  gulp.src('src/pages/**/*.{html,hbs,handlebars}')
    .pipe(panini({
      root: 'src/pages/',
      layouts: 'src/layouts/',
      partials: 'src/partials/',
      data: 'src/data/',
      helpers: 'src/helpers/'
    }))
    .pipe(gulp.dest('dist'));
});

gulp.task('pages:reset', function(cb) {
  panini.refresh();
  gulp.run('pages');
  cb();
});

gulp.task('styleguide', function(cb) {
  sherpa('src/styleguide/index.md', {
    output: 'dist/styleguide.html',
    template: 'src/styleguide/template.html'
  }, cb);
});

// Compile Sass into CSS
// In production, the CSS is compressed
gulp.task('sass', function() {
  var uncss = $.if(isProduction, $.uncss({
    html: ['src/**/*.html'],
    ignore: [
      new RegExp('^meta\..*'),
      new RegExp('^\.is-.*')
    ]
  }));

  var minifycss = $.if(isProduction, $.minifyCss());

  return gulp.src('src/assets/scss/app.scss')
    .pipe($.sourcemaps.init())
    .pipe($.sass({
      includePaths: PATHS.sass
    })
      .on('error', $.sass.logError))
    //Include external vendor CSS, such as Semantic UI
    .pipe(gulp.src(PATHS.external_css))
    .pipe($.concat('app.css'))
    .pipe($.autoprefixer({
      browsers: COMPATIBILITY
    }))
    .pipe(uncss)
    .pipe(minifycss)
    .pipe($.if(!isProduction, $.sourcemaps.write()))
    .pipe(gulp.dest('dist/assets/css'));
});

// Combine JavaScript into one file
// In production, the file is minified
gulp.task('javascript', function() {
  var uglify = $.if(isProduction, $.uglify()
    .on('error', function (e) {
      console.log(e);
    }));

  return gulp.src(PATHS.javascript)
    .pipe($.sourcemaps.init())
    .pipe($.concat('app.js'))
    .pipe(uglify)
    .pipe($.if(!isProduction, $.sourcemaps.write()))
    .pipe(gulp.dest('dist/assets/js'));
});

// Copy images to the "dist" folder
// In production, the images are compressed
gulp.task('images', function() {
  var imagemin = $.if(isProduction, $.imagemin({
    progressive: true
  }));

  return gulp.src('src/assets/img/**/*')
    .pipe(imagemin)
    .pipe(gulp.dest('dist/assets/img'));
});

// Build the "dist" folder by running all of the above tasks
gulp.task('build', function(done) {
  sequence('clean', ['pages', 'sass', 'javascript', 'images', 'copy'], 'styleguide', done);
});

// Start a server with LiveReload to preview the site in
gulp.task('server', ['build'], function() {
  browser.init({
    server: 'dist', port: PORT, tunnel: false
  });
});

// Build the site, run the server, and watch for file changes
gulp.task('default', ['build', 'server'], function() {
  gulp.watch(PATHS.assets, ['copy', browser.reload]);
  gulp.watch(['src/pages/**/*.html'], ['pages', browser.reload]);
  gulp.watch(['src/{layouts,partials}/**/*.html'], ['pages:reset', browser.reload]);
  gulp.watch(['src/assets/scss/**/*.scss'], ['sass', browser.reload]);
  gulp.watch(['src/assets/js/**/*.js'], ['javascript', browser.reload]);
  gulp.watch(['src/assets/img/**/*'], ['images', browser.reload]);
  gulp.watch(['src/styleguide/**'], ['styleguide', browser.reload]);
});
```

**Comment by neil-s** [comment]
Having read up a bit more about Gulp, I think this might be the issue: http://stackoverflow.com/a/24501930/181284. 

The new `gulp.src` line is replacing the previous stream of `.scss` files, with the new src of only the external CSS. I'm using Gulp 3.9.0, where this issue should have been fixed.

**Comment by gakimball** [comment]
Hm, well your other option, which isn't ideal, is to create a separate task that takes your compiled CSS and then merges it with the new files.

If you reference this task in `default` and `gulp.watch()`, it will always run `sass` first.

```js
gulp.task('css', ['sass'], function() {
  var FILES = [
    './dist/assets/css/app.css',
    './bower_components/...'
  ];

  gulp.src(FILES)
    .pipe($.concat('app.css'))
    .pipe(gulp.dest('./dist/assets/css');
});
```

**Comment by neil-s** [comment]
Actually, I realized that Sass successfully imports CSS files, even though a number of websites indicate that this is on the wishlist but hasn't been implemented yet. No need to change the file extension or anything, simply adding the path to the sass paths, and adding an import statement in `app.scss` was enough. What I was missing when I opened this issue was actually remembering to insert the import statement.
